app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentSheetController.kt

```diff
@@ -174,4 +255,56 @@ class ScheduleAppointmentSheetController @Inject constructor(
         .filter { user -> user.loggedInStatus == User.LoggedInStatus.UNAUTHORIZED }
         .map { { ui: Ui -> ui.finish() } }
   }
+
+  private fun generatePotentialAppointmentDate(
+      appointmentDate: LocalDate,
+      potentialAppointmentDates: List<PotentialAppointmentDate>
+  ): PotentialAppointmentDate {
+    val timeToAppointment = coerceTimeToAppointmentFromPotentialsForCalendarDate(potentialAppointmentDates, appointmentDate)
+    return PotentialAppointmentDate(appointmentDate, timeToAppointment)
+  }
+
+  private fun coerceTimeToAppointmentFromPotentialsForCalendarDate(
+      potentialAppointmentDates: List<PotentialAppointmentDate>,
+      date: LocalDate
+  ): TimeToAppointment {
+    val today = LocalDate.now(clock)
+    val exactMatchingTimeToAppointment = potentialAppointmentDates
+        .find { potentialAppointmentDate -> potentialAppointmentDate.scheduledFor == date }
+        ?.timeToAppointment
+
+    return exactMatchingTimeToAppointment ?: Days(ChronoUnit.DAYS.between(today, date).toInt())
+  }
+
+  private fun generatePotentialAppointmentDate(scheduleAppointmentIn: TimeToAppointment): PotentialAppointmentDate {
+    val today = LocalDate.now(clock)
+    return PotentialAppointmentDate(today.plus(scheduleAppointmentIn), scheduleAppointmentIn)
+  }
+
+  private fun generatePotentialAppointmentDates(scheduleAppointmentsIn: List<TimeToAppointment>): List<PotentialAppointmentDate> {
```

Comment:
We can simply deal with `TimeToAppointment` and create LocalDates only when we need them right? As mentioned before, the API of `PotentialAppointmentDate` allows for errors in input parameters passed.

Reply:
We can't do this because selecting the right `TimeToAppointment` when a calendar date is selected is not possible. We need to convert it to a `LocalDate` right at the beginning to make it more convenient.

Reply:
We ensure in tests that the `PotentialAppointment` is created with the same `TimeToAppointment` that the date is generated from. We will depend on manual testing to verify that the configuration itself is done correctly.

Reply:
We can do this. If you take a look at the commit I made just before this refactor, this is what was done. We computed the LocalDates finally only when they were required. Or at least LocalDate computation can also be made a property in `PotentialAppointmentDate`.

It is ok if you feel like skipping this change. We can come back to this design change in the future too.

Reply:
This is the commit in this branch you'll want to check out 42726e2233c802fc26c36ac9c55d32af668e12d4

Reply:
I intentionally did not do this because I felt the generation of dates everywhere from `ScheduleAppointment` just in time was becoming too complex and hard to follow.